# Exploring Full MovieLens Dataset

As the name suggests in this part we will start exploring Full MovieLens dataset for a better understanding of the data we are dealing with. It will serve as an introductory for creating the IMDB Top 250 Clone using Pandas library also known as the simplest recommender, while the rest of material will present the creation of different types of recommender systems e.g collaborative filtering, content-based systems, knwoledge-based and hybrid recommenders. 

The aforementioned dataset can be found at the following url: https://www.kaggle.com/rounakbanik/the-movies-dataset and contains metadata for all 45,000 movies listed in the Full MovieLens Dataset. The dataset consists of movies released on or before July 2017 containing information about budget, revenue, posters, release dates, languages, production companies, countries, TMDB vote counts, vote averages etc.

In this part we will focus on movies_metadata.csv file, and start analyzing data using pandas library to have a better understanding of the dataset we are dealing with. In the later chapters we will look as well on other datasets.

In [1]:
# Import required libraries
import pandas as pd

In [2]:
# Read the CSV file (dataset) into DataFrame
df = pd.read_csv('../datasets/movies_metadata.csv')

/home/dacy/.local/lib/python3.6/site-packages/IPython/core/interactiveshell.py:3049: DtypeWarning: Columns (10) have mixed types. Specify dtype option on import or set low_memory=False.
  interactivity=interactivity, compiler=compiler, result=result)


In [3]:
# Read the first 5 movies or occurencies
df.head()

adult                              belongs_to_collection    budget  \
0  False  {'id': 10194, 'name': 'Toy Story Collection', ...  30000000   
1  False                                                NaN  65000000   
2  False  {'id': 119050, 'name': 'Grumpy Old Men Collect...         0   
3  False                                                NaN  16000000   
4  False  {'id': 96871, 'name': 'Father of the Bride Col...         0   

                                              genres  \
0  [{'id': 16, 'name': 'Animation'}, {'id': 35, '...   
1  [{'id': 12, 'name': 'Adventure'}, {'id': 14, '...   
2  [{'id': 10749, 'name': 'Romance'}, {'id': 35, ...   
3  [{'id': 35, 'name': 'Comedy'}, {'id': 18, 'nam...   
4                     [{'id': 35, 'name': 'Comedy'}]   

                               homepage     id    imdb_id original_language  \
0  http://toystory.disney.com/toy-story    862  tt0114709                en   
1                                   NaN   8844  tt0113497                en   
2                                   NaN  15602  tt0113228                en   
3                                   NaN  31357  tt0114885                en   
4                                   NaN  11862  tt0113041                en   

                original_title  \
0                    Toy Story   
1                      Jumanji   
2             Grumpier Old Men   
3            Waiting to Exhale   
4  Father of the Bride Part II   

                                            overview    ...     release_date  \
0  Led by Woody, Andy's toys live happily in his ...    ...       1995-10-30   
1  When siblings Judy and Peter discover an encha...    ...       1995-12-15   
2  A family wedding reignites the ancient feud be...    ...       1995-12-22   
3  Cheated on, mistreated and stepped on, the wom...    ...       1995-12-22   
4  Just when George Banks has recovered from his ...    ...       1995-02-10   

       revenue runtime                                   spoken_languages  \
0  373554033.0    81.0           [{'iso_639_1': 'en', 'name': 'English'}]   
1  262797249.0   104.0  [{'iso_639_1': 'en', 'name': 'English'}, {'iso...   
2          0.0   101.0           [{'iso_639_1': 'en', 'name': 'English'}]   
3   81452156.0   127.0           [{'iso_639_1': 'en', 'name': 'English'}]   
4   76578911.0   106.0           [{'iso_639_1': 'en', 'name': 'English'}]   

     status                                            tagline  \
0  Released                                                NaN   
1  Released          Roll the dice and unleash the excitement!   
2  Released  Still Yelling. Still Fighting. Still Ready for...   
3  Released  Friends are the people who let you be yourself...   
4  Released  Just When His World Is Back To Normal... He's ...   

                         title  video vote_average vote_count  
0                    Toy Story  False          7.7     5415.0  
1                      Jumanji  False          6.9     2413.0  
2             Grumpier Old Men  False          6.5       92.0  
3            Waiting to Exhale  False          6.1       34.0  
4  Father of the Bride Part II  False          5.7      173.0  

[5 rows x 24 columns]

We can see that our dataset contains 24 columns or features where each row representing a movie. To understand better our data we will make use of the Pandas built-in functions and methods.

In [4]:
# Output the shape of df
df.shape

(45466, 24)

Using df.shape we can see that we are dealing with over 45000 movie records. We can also see that we have 24 columns. Each column represents a feature or a piece of metadata about the movie.

To understand the features we are working with, we run the following code:

In [5]:
# Output the columns of df
df.columns

Index(['adult', 'belongs_to_collection', 'budget', 'genres', 'homepage', 'id',
       'imdb_id', 'original_language', 'original_title', 'overview',
       'popularity', 'poster_path', 'production_companies',
       'production_countries', 'release_date', 'revenue', 'runtime',
       'spoken_languages', 'status', 'tagline', 'title', 'video',
       'vote_average', 'vote_count'],
      dtype='object')

We can see that we have a lot of information on these movies, including their title, budget, genres, release date, and revenue. Next, we will start accessing a particular movie. The first way to do this is by using .iloc method.
This will allow us to select rows based on the numeric position (starting from zero).

In [6]:
# To access the second movie in the DataFrame
df.iloc[1]

adult                                                                False
belongs_to_collection                                                  NaN
budget                                                            65000000
genres                   [{'id': 12, 'name': 'Adventure'}, {'id': 14, '...
homepage                                                               NaN
id                                                                    8844
imdb_id                                                          tt0113497
original_language                                                       en
original_title                                                     Jumanji
overview                 When siblings Judy and Peter discover an encha...
popularity                                                         17.0155
poster_path                               /vzmL6fP7aPKNKPRTFnZmiUfciyV.jpg
production_companies     [{'name': 'TriStar Pictures', 'id': 559}, {'na...
production_countries     

The output gives us information about the movie on each of its 24 features. We can see that the accessed movie on second position refers to 'Jumanji' which was released on 1995-12-15 among other things.

The second way to do it is by accessing the DataFrame index. In order to do that we need to set the index to the tile of the movie and try to access 'Jumanji' using its this index.

In [7]:
# Change the index to the title
df = df.set_index('title')

In [8]:
# Access the movie with title 'Jumanji'
df.loc['Jumanji']

adult                                                                False
belongs_to_collection                                                  NaN
budget                                                            65000000
genres                   [{'id': 12, 'name': 'Adventure'}, {'id': 14, '...
homepage                                                               NaN
id                                                                    8844
imdb_id                                                          tt0113497
original_language                                                       en
original_title                                                     Jumanji
overview                 When siblings Judy and Peter discover an encha...
popularity                                                         17.0155
poster_path                               /vzmL6fP7aPKNKPRTFnZmiUfciyV.jpg
production_companies     [{'name': 'TriStar Pictures', 'id': 559}, {'na...
production_countries     

Using .loc we should see an output identical to the previous cell. We will revert it back to our zero-based numeric index:


In [9]:
# Revert back to the previous zero-based index
df = df.reset_index()

In order to make a simple analysis on our dataset, we will create a smaller DataFrame with fewer columns. The given DataFrame will contain data on the following features: title, release_date, budget, revenue, runtime, and genres

In [10]:
# Create a smaller DataFrame with the following features: title, release_date, budget, revenue, runtime, genres
movie_df = df[['title', 'release_date', 'budget', 'revenue', 'runtime', 'genres']]

In [11]:
# Read the first 5 occurencies
movie_df.head()

title release_date    budget      revenue  runtime  \
0                    Toy Story   1995-10-30  30000000  373554033.0     81.0   
1                      Jumanji   1995-12-15  65000000  262797249.0    104.0   
2             Grumpier Old Men   1995-12-22         0          0.0    101.0   
3            Waiting to Exhale   1995-12-22  16000000   81452156.0    127.0   
4  Father of the Bride Part II   1995-02-10         0   76578911.0    106.0   

                                              genres  
0  [{'id': 16, 'name': 'Animation'}, {'id': 35, '...  
1  [{'id': 12, 'name': 'Adventure'}, {'id': 14, '...  
2  [{'id': 10749, 'name': 'Romance'}, {'id': 35, ...  
3  [{'id': 35, 'name': 'Comedy'}, {'id': 18, 'nam...  
4                     [{'id': 35, 'name': 'Comedy'}]

We can see that the newly generated DataFrame contains movie data regarding: title, release_date, budget, revenue, runtime, genres

Next we will take a look at the data types of our various features

In [12]:
# Check data types for each of the features (column names)
movie_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 45466 entries, 0 to 45465
Data columns (total 6 columns):
title           45460 non-null object
release_date    45379 non-null object
budget          45466 non-null object
revenue         45460 non-null float64
runtime         45203 non-null float64
genres          45466 non-null object
dtypes: float64(2), object(4)
memory usage: 2.1+ MB


Looking at each of the feature we can see that 'revenue' and 'runtime' are of type float64 while the rest of them are of type object. However we can change the type for the 'budget' feature to float data type.

In [13]:
# Convert budget to float
# df['budget'] = df['budget'].astype('float')

By running the precedent code, the cell will throw a ValuError. From the given error we understand that one of the record or budget fields had a '/zasf...' string as its value. Therefore, pandas was unable to convert it into a floating number.

To solve this problem, we will use the apply() method. This will allow us to apply a function to every field in a particular column and convert it into a return value. We will convert every number field in 'budget' to float and if it fails then we will convert it to a NaN value.

In [14]:
# Import required library
import numpy as np

In [15]:
# Function to convert to float manually
def to_float(x):
    try:
        x = float(x)
    except:
        x = np.nan
    return x

In [16]:
# Apply to_float function to all values in the budget column
movie_df['budget'] = movie_df['budget'].apply(to_float)

/home/dacy/.local/lib/python3.6/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  


In [17]:
# Try converting to float using pandas astype
movie_df['budget'] = movie_df['budget'].astype('float')

/home/dacy/.local/lib/python3.6/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  


In [18]:
# Check for the last change
movie_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 45466 entries, 0 to 45465
Data columns (total 6 columns):
title           45460 non-null object
release_date    45379 non-null object
budget          45463 non-null float64
revenue         45460 non-null float64
runtime         45203 non-null float64
genres          45466 non-null object
dtypes: float64(3), object(3)
memory usage: 2.1+ MB


We can see that the 'budget' feature has been changed from object data type to float64.

We will define a new feature called 'year' that represents the year of release. The recommended way to do this would be by using the datetime functionality that pandas gives us:

In [19]:
# Convert release_date into pandas datetime format
movie_df['release_date'] = pd.to_datetime(movie_df['release_date'], errors='coerce')

/home/dacy/.local/lib/python3.6/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  


In [20]:
# Check if release_date is changed from object to datetime
movie_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 45466 entries, 0 to 45465
Data columns (total 6 columns):
title           45460 non-null object
release_date    45376 non-null datetime64[ns]
budget          45463 non-null float64
revenue         45460 non-null float64
runtime         45203 non-null float64
genres          45466 non-null object
dtypes: datetime64[ns](1), float64(3), object(2)
memory usage: 2.1+ MB


In [21]:
# Extract 'year' from datetime
movie_df['year'] = movie_df['release_date'].apply(lambda x: str(x).split('-')[0] if x != np.nan else np.nan)

/home/dacy/.local/lib/python3.6/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  


In [22]:
# Display the DataFrame with the new 'year' feature
movie_df.head()

title release_date      budget      revenue  runtime  \
0                    Toy Story   1995-10-30  30000000.0  373554033.0     81.0   
1                      Jumanji   1995-12-15  65000000.0  262797249.0    104.0   
2             Grumpier Old Men   1995-12-22         0.0          0.0    101.0   
3            Waiting to Exhale   1995-12-22  16000000.0   81452156.0    127.0   
4  Father of the Bride Part II   1995-02-10         0.0   76578911.0    106.0   

                                              genres  year  
0  [{'id': 16, 'name': 'Animation'}, {'id': 35, '...  1995  
1  [{'id': 12, 'name': 'Adventure'}, {'id': 14, '...  1995  
2  [{'id': 10749, 'name': 'Romance'}, {'id': 35, ...  1995  
3  [{'id': 35, 'name': 'Comedy'}, {'id': 18, 'nam...  1995  
4                     [{'id': 35, 'name': 'Comedy'}]  1995

We can have a deepen understanding of which are the oldest movies available in the dataset? To answer the question, we can sort the DataFrame based on the year of release:

In [23]:
# Sort DataFrame based on release year
movie_df = movie_df.sort_values('year')

In [24]:
movie_df.head(10)

title release_date  budget  revenue  runtime  \
34940               Passage of Venus   1874-12-09     0.0      0.0      1.0   
34937     Sallie Gardner at a Gallop   1878-06-14     0.0      0.0      1.0   
41602                Buffalo Running   1883-11-19     0.0      0.0      1.0   
34933    Man Walking Around a Corner   1887-08-18     0.0      0.0      1.0   
34938  Traffic Crossing Leeds Bridge   1888-10-15     0.0      0.0      1.0   
34934               Accordion Player   1888-01-01     0.0      0.0      1.0   
34936            Monkeyshines, No. 2   1890-11-21     0.0      0.0      1.0   
39954            Monkeyshines, No. 3   1890-11-21     0.0      0.0      1.0   
34935            Monkeyshines, No. 1   1890-11-21     0.0      0.0      1.0   
34939      London's Trafalgar Square   1890-01-01     0.0      0.0      1.0   

                                    genres  year  
34940  [{'id': 99, 'name': 'Documentary'}]  1874  
34937  [{'id': 99, 'name': 'Documentary'}]  1878  
41602  [{'id': 99, 'name': 'Documentary'}]  1883  
34933  [{'id': 99, 'name': 'Documentary'}]  1887  
34938  [{'id': 99, 'name': 'Documentary'}]  1888  
34934  [{'id': 99, 'name': 'Documentary'}]  1888  
34936                                   []  1890  
39954                                   []  1890  
34935       [{'id': 35, 'name': 'Comedy'}]  1890  
34939  [{'id': 99, 'name': 'Documentary'}]  1890

Looking at the dataset we can see that the oldest movie is 'Passage of Venus' which was released on 1874-12-09.

We can even find which were the most successful movies of all time. To do this we'll use the sort_values() method with an additional asceding=False parameter to sort DataFrame in descending order

In [25]:
movie_df = movie_df.sort_values('revenue', ascending=False)

In [26]:
movie_df.head()

title release_date       budget       revenue  \
14551                        Avatar   2009-12-10  237000000.0  2.787965e+09   
26555  Star Wars: The Force Awakens   2015-12-15  245000000.0  2.068224e+09   
1639                        Titanic   1997-11-18  200000000.0  1.845034e+09   
17818                  The Avengers   2012-04-25  220000000.0  1.519558e+09   
25084                Jurassic World   2015-06-09  150000000.0  1.513529e+09   

       runtime                                             genres  year  
14551    162.0  [{'id': 28, 'name': 'Action'}, {'id': 12, 'nam...  2009  
26555    136.0  [{'id': 28, 'name': 'Action'}, {'id': 12, 'nam...  2015  
1639     194.0  [{'id': 18, 'name': 'Drama'}, {'id': 10749, 'n...  1997  
17818    143.0  [{'id': 878, 'name': 'Science Fiction'}, {'id'...  2012  
25084    124.0  [{'id': 28, 'name': 'Action'}, {'id': 12, 'nam...  2015

From the output given we conclude that the most succesfull released movie was 'Avatar' which a revenue of 2.78 $billion.

We can filter out movies from the DataFrame that satisfies a certain condition or criteria. For instance, we only want movies that earned more than $1 billion. Pandas makes this possible using its Boolean indexing feature.

In [27]:
# Get all movies that earned more than $1 billion
movie_dataset1 = movie_df[movie_df['revenue'] > 1e9]

In [28]:
movie_dataset1

title release_date  \
14551                                         Avatar   2009-12-10   
26555                   Star Wars: The Force Awakens   2015-12-15   
1639                                         Titanic   1997-11-18   
17818                                   The Avengers   2012-04-25   
25084                                 Jurassic World   2015-06-09   
28830                                      Furious 7   2015-04-01   
26558                        Avengers: Age of Ultron   2015-04-22   
17437   Harry Potter and the Deathly Hallows: Part 2   2011-07-07   
22110                                         Frozen   2013-11-27   
42222                           Beauty and the Beast   2017-03-16   
43255                        The Fate of the Furious   2017-04-12   
20830                                     Iron Man 3   2013-04-18   
30700                                        Minions   2015-06-17   
26567                     Captain America: Civil War   2016-04-27   
17293                 Transformers: Dark of the Moon   2011-06-28   
7000   The Lord of the Rings: The Return of the King   2003-12-01   
19261                                        Skyfall   2012-10-25   
23617                Transformers: Age of Extinction   2014-06-25   
18252                          The Dark Knight Rises   2012-07-16   
15348                                    Toy Story 3   2010-06-16   
11008     Pirates of the Caribbean: Dead Man's Chest   2006-06-20   
41489                   Rogue One: A Star Wars Story   2016-12-14   
17124    Pirates of the Caribbean: On Stranger Tides   2011-05-14   
38176                                   Finding Dory   2016-06-16   
14892                            Alice in Wonderland   2010-03-03   
36253                                       Zootopia   2016-02-11   
19971              The Hobbit: An Unexpected Journey   2012-11-26   
44009                                Despicable Me 3   2017-06-15   
12481                                The Dark Knight   2008-07-16   

            budget       revenue  runtime  \
14551  237000000.0  2.787965e+09    162.0   
26555  245000000.0  2.068224e+09    136.0   
1639   200000000.0  1.845034e+09    194.0   
17818  220000000.0  1.519558e+09    143.0   
25084  150000000.0  1.513529e+09    124.0   
28830  190000000.0  1.506249e+09    137.0   
26558  280000000.0  1.405404e+09    141.0   
17437  125000000.0  1.342000e+09    130.0   
22110  150000000.0  1.274219e+09    102.0   
42222  160000000.0  1.262886e+09    129.0   
43255  250000000.0  1.238765e+09    136.0   
20830  200000000.0  1.215440e+09    130.0   
30700   74000000.0  1.156731e+09     91.0   
26567  250000000.0  1.153304e+09    147.0   
17293  195000000.0  1.123747e+09    154.0   
7000    94000000.0  1.118889e+09    201.0   
19261  200000000.0  1.108561e+09    143.0   
23617  210000000.0  1.091405e+09    165.0   
18252  250000000.0  1.084939e+09    165.0   
15348  200000000.0  1.066970e+09    103.0   
11008  200000000.0  1.065660e+09    151.0   
41489  200000000.0  1.056057e+09    133.0   
17124  380000000.0  1.045714e+09    136.0   
38176  200000000.0  1.028571e+09     97.0   
14892  200000000.0  1.025491e+09    108.0   
36253  150000000.0  1.023784e+09    108.0   
19971  250000000.0  1.021104e+09    169.0   
44009   80000000.0  1.020063e+09     96.0   
12481  185000000.0  1.004558e+09    152.0   

                                                  genres  year  
14551  [{'id': 28, 'name': 'Action'}, {'id': 12, 'nam...  2009  
26555  [{'id': 28, 'name': 'Action'}, {'id': 12, 'nam...  2015  
1639   [{'id': 18, 'name': 'Drama'}, {'id': 10749, 'n...  1997  
17818  [{'id': 878, 'name': 'Science Fiction'}, {'id'...  2012  
25084  [{'id': 28, 'name': 'Action'}, {'id': 12, 'nam...  2015  
28830                     [{'id': 28, 'name': 'Action'}]  2015  
26558  [{'id': 28, 'name': 'Action'}, {'id': 12, 'nam...  2015  
17437  [{'id': 10751, 'name': 'Family'}, {'id': 14, '...  2011  
22110  [{'id': 16, 'name': 'Animation'},

It is possible to apply even multiple conditions. For instance, let's say we only wanted movies that earned more than 
$1 billion, but where the budget less than $150 million.

In [29]:
movie_dataset2 = movie_df[(movie_df['revenue'] > 1e9) & (movie_df['budget'] < 1.5e8)]

In [30]:
movie_dataset2

title release_date  \
17437   Harry Potter and the Deathly Hallows: Part 2   2011-07-07   
30700                                        Minions   2015-06-17   
7000   The Lord of the Rings: The Return of the King   2003-12-01   
44009                                Despicable Me 3   2017-06-15   

            budget       revenue  runtime  \
17437  125000000.0  1.342000e+09    130.0   
30700   74000000.0  1.156731e+09     91.0   
7000    94000000.0  1.118889e+09    201.0   
44009   80000000.0  1.020063e+09     96.0   

                                                  genres  year  
17437  [{'id': 10751, 'name': 'Family'}, {'id': 14, '...  2011  
30700  [{'id': 10751, 'name': 'Family'}, {'id': 16, '...  2015  
7000   [{'id': 12, 'name': 'Adventure'}, {'id': 14, '...  2003  
44009  [{'id': 28, 'name': 'Action'}, {'id': 16, 'nam...  2017

As we can see only four movies make it into a list.

Next, we will take a look at the 'runtime' feature from the dataset and make a simple analysis on it. For example we can look at the shortest and longest-running movies of all time. 

In [31]:
# Get the runtime Series object
runtime = movie_df['runtime']

In [32]:
# Get the longest runtime of any movie
runtime.max()

1256.0

In [33]:
# Get the shortest runtime of any movie
runtime.min()

0.0

We can see that the longest movie runtime is more than 1256 minutes in lenghet while the shortest movie runtime is 0. Looking at the generated result indicates a strange result and needs a deeper inspection on that.

It is possible to calculate the mean and median of the Series in this way. 

In [35]:
# Get the budget Series object
budget = movie_df['budget']

In [36]:
# Print the mean budget of the movies
budget.mean()

4224578.813474693

In [37]:
# Print the median budget of the movies
budget.median()

0.0

Looking at the generated results we can see that the average budget of a movie is $4.2 million and the median budget is 0!. This suggests that at least half of the movies in our dataset have no budget at all. Like in the previous case, strange results deman closer inspection. It is highly likely that a zero budget indicates that the data is not available.

What is the revenue that the 90th-percentile movie generated? We can make use of the quantile function in order to determine this

In [39]:
# Get the revenue Series object
revenue = movie_df['revenue']

In [40]:
# Revenue generated by the 90th percentile movie
revenue.quantile(0.90)

8267610.399999982

Doing so quantile on the 'revenue' column, we get a result of $8.26 million. This indicates that only 10\% of the movies in our dataset earned more than 8.26 million in revenue.

Let's find out the number of movies released each year. We do this using the value_counts() method on 'year' feature from the DataFrame

In [41]:
# Get number of movies released each year
movie_df['year'].value_counts()

2014    1974
2015    1905
2013    1889
2012    1722
2011    1667
2016    1604
2009    1586
2010    1501
2008    1473
2007    1320
2006    1270
2005    1125
2004     992
2002     905
2003     882
2001     865
2000     789
1999     723
1998     722
1997     661
1996     633
1995     599
1994     544
2017     532
1993     489
1988     467
1987     462
1992     453
1989     439
1990     427
        ... 
1912      14
1896      14
1900      14
1894      13
1898      13
1913      13
1901      10
1909      10
1897       8
1904       8
1895       7
1907       7
1910       7
1906       7
1911       6
1891       6
1908       6
2018       5
1899       5
1905       5
1890       5
1892       3
1888       2
1902       2
1874       1
1893       1
1883       1
1887       1
1878       1
2020       1
Name: year, Length: 136, dtype: int64

The generated result shows that in six years from our dataset we have only one movie as record, including (2020) while in 2014 we have the highest number with 1974 movies released.